src/native/managed/cdacreader/src/Contracts/RuntimeTypeSystem_1.cs

```diff
@@ -154,4 +157,184 @@ public uint GetTypeDefToken(MethodTableHandle methodTableHandle)
 
     public bool IsDynamicStatics(MethodTableHandle methodTableHandle) => _methodTables[methodTableHandle.Address].Flags.IsDynamicStatics;
 
+    public ReadOnlySpan<MethodTableHandle> GetInstantiation(MethodTableHandle methodTableHandle)
+    {
+        MethodTable methodTable = _methodTables[methodTableHandle.Address];
+        if (!methodTable.Flags.HasInstantiation)
+            return default;
+
+        TargetPointer perInstInfo = methodTable.PerInstInfo;
+        TargetPointer genericsDictInfo = perInstInfo - (ulong)_target.PointerSize;
+
+        TargetPointer dictionaryPointer = _target.ReadPointer(perInstInfo);
+
+        int numberOfGenericArgs = _target.ProcessedData.GetOrAdd<GenericsDictInfo>(genericsDictInfo).NumTypeArgs;
+        MethodTableArray instantiation = _target.ProcessedData.GetOrAdd<(TargetPointer, int), MethodTableArray>
+            ((dictionaryPointer, numberOfGenericArgs));
+
+        return instantiation.Types.AsSpan();
```

Comment:
why does the key for the data include the number of generic args?  can the same `dictionaryPointer` be viewed with a different number of generic args?

Reply:
Because just the pointer to the start of the data doesn't quite encode enough to identify the number of generic args. I'm going to restructure this so that instead of this being a general purpose `TypeHandleArray`... that it becomes a `TypeInstantiation` type, that type is defined in the contract implementation, and the pointer that makes identifies it is the `MethodTable` pointer. That should make this simpler, remove the extra key concept, remove the concern about Data contracts knowing about contracts, and make it so that repeated calls to `GetInstantiation` api are just the cost of doing data lookup, which should help us put our new design on a path to actually be *faster* than the old model.